### DISCUSSION TITLE
[2023-10-02] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by wdanilo** [enso-org/enso#7944] [Weekly Demo Day]
I was working on website. Notable changes:

1. We have now nodes and component browser, fully animatable. I did not yet connect James filtering algorithm, but I will do it today.
2. There is a new "animation engine" allowing animations to run without user scrolling. These animations are automatically revertible and they can automatically run faster, e.g. after user scrolls the web page up fast.
3. I'm working now on argument pop-ups and node connections. After that's done, we can just add nodes and animations with "single line of code". This should make the website showable this week.

Planned work:
1. Finishing these animations. They already took too much time, CSS sucks so much sometimes.
2. Integrating with website pricing and chat – some help from E-Hern would be greatly appreciated.

https://github.com/enso-org/enso/assets/1623053/a7745caa-9ebc-48eb-9132-2aaf2fd08583

**Comment by MichaelMauderer** [comment]
Working on [Apply expression updates #7783](https://github.com/enso-org/enso/issues/7783)
* delayed by #7898

**Comment by vitvakatu** [comment]
Last week I was fighting various build issues (recent `rust-analyzer` not working with the project anymore, `gui2` broke `run watch`, and apparently cleaning all local caches was a bad idea).

Still working on a few issues with the old IDE:
- https://github.com/enso-org/enso/pull/7779 – under investigation, blocked by QA issues.
- https://github.com/enso-org/enso/pull/7844 – almost fixed.

Finished support for group colors in gui2. Now we generate `oklch` colors, and also support fallback to `sRGB` in case the browser is not supporting fancy colorspaces.
- https://github.com/enso-org/enso/pull/7902


https://github.com/enso-org/enso/assets/6566674/7177f041-93e9-4536-b19f-6b67a2da8b7e

**Comment by mwu-tow** [comment]
* Work on support for the new GUI and IDE2: delayed, PR under testing; (https://github.com/enso-org/enso/pull/7832)
* Fixes for AG Grid License (https://github.com/enso-org/enso/pull/7914)
* CI support for Standard Library S3 tests (https://github.com/enso-org/enso/pull/7920)

**Comment by kazcw** [comment]
Finishing:
- #7701
Working on:
- #7864

**Comment by somebody1234** [comment]
- Shortcuts and node selection (writing new shortcuts API, fixing bugs with selections) (merged): #7893
- Visualization updates: #7873
- (today) Begin work on demo scenes (using Histoire as suggested): #7945


https://github.com/enso-org/enso/assets/4046547/6f413c87-fbb4-4db0-9956-d8924c812360

**Comment by hubertp** [comment]
Relevant updates:
- Investigating issues between LS and new GUI #7898 
- Opening projects using packaged engine #7868
